repairnator/repairnator-pipeline/src/main/java/fr/inria/spirals/repairnator/process/step/pathes/ComputeTestDir.java

```diff
@@ -129,23 +130,22 @@ private void computeMetricsOnTest(File[] sources) {
                 int nbFile = FileUtils.listFiles(f, new String[] {"java"}, true).size();
                 totalAppFiles += nbFile;
             }
-            this.inspector.getJobStatus().getMetrics().setNbFileTests(totalAppFiles);
+            this.getInspector().getJobStatus().getMetrics().setNbFileTests(totalAppFiles);
         }
     }
 
     @Override
-    protected void businessExecute() {
+    protected StepStatus businessExecute() {
         this.getLogger().debug("Computing the test directory ...");
-        File[] sources = this.searchForSourcesDirectory(this.inspector.getRepoLocalPath(), true);
+        File[] sources = this.searchForSourcesDirectory(this.getInspector().getRepoLocalPath(), true);
 
-        this.getInspector().getJobStatus().setTestDir(sources);
-        this.computeMetricsOnTest(sources);
-
-        if (sources == null) {
+        if (sources == null || sources.length == 0) {
             this.addStepError("Fail to find the tests directory.");
-            this.setPipelineState(PipelineState.TESTDIRNOTCOMPUTED);
+            return StepStatus.buildError(this, PipelineState.TESTDIRNOTCOMPUTED);
```

Comment:
`this.getInspector().getJobStatus().setTestDir(null);`?

Reply:
By default it's already null. I don't want to set explicitely a null value.